diesel/src/r2d2.rs

```diff
@@ -171,7 +168,70 @@ where
     }
 
     fn transaction_manager(&self) -> &Self::TransactionManager {
-        (&**self).transaction_manager()
+        // This is actually fine because we have an #[repr(transparent)]
+        // on LoggingTransactionManager, which means the layout is the same
+        // as the inner type
+        // See the ref-cast crate for a longer version: https://github.com/dtolnay/ref-cast
+        unsafe {
+            &*((&**self).transaction_manager() as *const _ as *const Self::TransactionManager)
+        }
+    }
```

Comment:
As discussed on Slack this should also override `begin_test_transaction` (and any other method from `Connection`)

(I will try to fix that in diesel-master as well)